code/modules/projectiles/projectile/special/curse.dm

```diff
@@ -48,9 +47,10 @@
 	for(var/obj/effect/temp_visual/dir_setting/curse/grasp_portal/G in starting)
 		qdel(G)
 	new /obj/effect/temp_visual/dir_setting/curse/grasp_portal/fading(starting, dir)
-	var/datum/beam/D = starting.Beam(T, icon_state = "curse[handedness]", time = 32, maxdistance = INFINITY, beam_type=/obj/effect/ebeam/curse_arm, beam_sleep_time = 1)
+	var/datum/beam/D = starting.Beam(T, icon_state = "curse[handedness]", beam_type=/obj/effect/ebeam/curse_arm)
 	for(var/b in D.elements)
 		var/obj/effect/ebeam/B = b
 		animate(B, alpha = 0, time = 32)
+		QDEL_IN(B, 32)
```

Comment:
Why do this on B rather then D?

Reply:
Better question: why doesn't this use the new beam visuals?!